yt/loaders.py

```diff
@@ -1268,90 +1268,102 @@ def flatten(l):
     """
-    Load sample data with yt. Simple wrapper around yt.load to include fetching
+    Load sample data with yt. Simple wrapper around `yt.load` to include fetching
     data with pooch.
 
+    The data registry table can be retrieved and visualized using
+    `yt.sample_data.api.get_data_registry_table`.
+
+    This function requires pandas and pooch to be installed.
+
     Parameters
     ----------
-    fn : str or None
-        The name of the sample data to load. This is generally the name of the
-        folder of the dataset. For IsolatedGalaxy, the name would be
-        `IsolatedGalaxy`.  If `None` is supplied, the return value
-        will be a list of all known datasets (by name).
-
-    specific_file : str, optional
-        optional argument -- the name of the file to load that is located
-        within sample dataset of `name`. For the dataset `enzo_cosmology_plus`,
-        which has a number of timesteps available, one may wish to choose
-        DD0003. The file specifically would be
-        `enzo_cosmology_plus/DD0003/DD0003`, and the argument passed to this
-        variable would be `DD0003/DD0003`
+    fn : str
+        The `filename` of the dataset to load, as defined in the data registry
+        table.
 
     pbar: bool
         display a progress bar
 
+    Any additional keyword argument is passed down to `yt.load`.
+    Note that in case of collision with predefined keyword arguments as set in
+    the data registry, the ones passed to this function take priority.
     """
 
-    fido = PoochHandle()
+    from yt.sample_data.api import (
+        _download_sample_data_file,
+        _get_archive_filename,
+        _get_test_data_dir_path,
+        get_data_registry_table,
+        lookup_on_disk_data,
+    )
 
-    if fn is None:
-        keys = []
-        for key in fido._registry:
-            for ext in _extensions_to_strip:
-                if key.endswith(ext):
-                    key = key[: -len(ext)]
-            keys.append(key)
-        return keys
+    pooch_logger = pooch.utilsget_logger()
 
-    base_path = fido.pooch_obj.path
+    topdir, _, specific_file = str(fn).partition(os.path.sep)
 
-    registered_fname, name, extension = fido._validate_sample_fname(
-        fn
-    )  # todo: make this part of the class
+    registry_table = get_data_registry_table()
+    # PR 3089
+    # note: in the future the registry table should be reindexed
+    # so that the following line can be replaced with
+    #
+    # specs = registry_table.loc[fn]
+    #
+    # however we don't want to do it right now because the "filename" column is
+    # currently incomplete
+    specs = registry_table.query(f"`filename` == '{topdir}'").iloc[0]
 
-    downloader = None
-    if pbar:
-        downloader = pooch.pooch.HTTPDownloader(progressbar=True)
+    if not specs["load_name"]:
+        raise ValueError(
+            "Registry appears to be corrupted: could not find a 'load_name' entry for this dataset."
```

Comment:
oh but _is_ currently corrupted: we're missing many metadata points, making this feature inherently broken for a number of sample datasets.
see for instance
https://github.com/yt-project/yt/pull/3118 (still not fixed)
or
https://github.com/yt-project/yt/pull/3121

The reason this error is not "could not find ..." is that the line _before_ that will already error out in cas the requested dataset isn't findable, but we still need this check because recognising the entry doesn't mean we're able to load (currently).